agent/structs/intention_test.go

```diff
@@ -453,3 +454,9 @@ func TestIntention_String(t *testing.T) {
 		})
 	}
 }
+
+func TestIntentionQueryRequest_CacheInfoKey(t *testing.T) {
+	// TODO: should these fields be included in the key?
+	ignored := []string{"IntentionID", "Check", "Exact", "QueryOptions"}
```

Comment:
IntentionID, Check, and Exact should also be included. Specifying these different fields leads to different types of queries. It looks like the existing CacheInfo is from when intentions were first added.